src/expr/src/expr/array.rs

```diff
@@ -0,0 +1,206 @@
+        } else {
+            None
+        };
+
+        Ok(Self {
+            array,
+            delimiter,
+            null_string,
+        })
+    }
+}
+
+impl Expression for ArrayToStringExpression {
+    fn return_type(&self) -> DataType {
+        DataType::Varchar
+    }
+
+    fn eval(&self, input: &DataChunk) -> Result<ArrayRef> {
+        let list_array = self.array.eval_checked(input)?;
+        let list_array = list_array.as_list();
+
+        let delim_array = self.delimiter.eval_checked(input)?;
+        let delim_array = delim_array.as_utf8();
+
+        let null_string_array = if let Some(expr) = &self.null_string {
+            let null_string_array = expr.eval_checked(input)?;
+            Some(null_string_array)
+        } else {
+            None
+        };
+
+        let mut output = Utf8ArrayBuilder::with_meta(input.capacity(), ArrayMeta::Simple);
+
+        for (i, vis) in input.vis().iter().enumerate() {
+            if !vis {
+                output.append_null();
+            } else {
+                let array = list_array.value_at(i);
+                let delim = delim_array.value_at(i);
+                let null_string = if let Some(a) = &null_string_array {
+                    a.as_utf8().value_at(i)
+                } else {
+                    None
+                };
+
+                if let Some(array) = array && let Some(delim) = delim {
+                    let result =
+                    if let Some(null_string) = null_string {
+                    Self::evaluate_with_nulls(array, delim, null_string)
+                    } else {
+                        Self::evaluate(array, delim)
+                    };
+                    output.append(Some(&result));
+                } else {
+                    output.append_null();
+                }
+            }
+        }
+        Ok(Arc::new(output.finish().into()))
+    }
+
+    fn eval_row(&self, input: &OwnedRow) -> Result<Datum> {
+        let array = self.array.eval_row(input)?;
+        let delimiter = self.delimiter.eval_row(input)?;
+
+        let result = if let Some(array) = array && let Some(delimiter) = delimiter {
+            if let Some(e) = &self.null_string {
+                let null_string = e.eval_row(input)?;
+                if let Some(null_string) = null_string {
+                    Some(Self::evaluate_with_nulls(array.as_scalar_ref_impl().into_list(), delimiter.as_utf8(), null_string.as_utf8()))
+                } else {
+                    Some(Self::evaluate(array.as_scalar_ref_impl().into_list(), delimiter.as_utf8()))
+                }
+            } else {
+            None
+            }
+        } else{
+            None
+        };
+        Ok(result.map(|r| r.into()))
+    }
+}
+
+impl ArrayToStringExpression {
+    fn evaluate(array: ListRef<'_>, delimiter: &str) -> String {
+        array
+            .values_ref()
+            .iter()
+            .flat_map(|f| f.iter())
+            .map(|f| f.to_text())
```

Comment:
Consider using the `ToText::write` API to avoid generating Strings for better performance.